### ISSUE TITLE
Running ketos train on 4.2.0 with an empty line in a compiled arrow raises an exception

### ISSUE THREAD

**Issue by PonteIneptique** [mittagessen/kraken#389]
```
│ .krakenv/lib/python3.8/site-packages/kraken/lib/dataset.py:587 in                  │
│ _apply_text_transform                                                                            │
│                                                                                                  │
│    584 │   │   if not text:                                                                      │
│    585 │   │   │   logger.debug(f'Text line "{sample["text"]}" is empty after transformations')  │
│    586 │   │   │   if not self.ignore_empty_lines:                                               │
│ ❱  587 │   │   │   │   raise Exception('empty text line')                                        │
│    588 │   │   return text      
```

It seems like a flag for ignoring this would be ok

**Comment by PonteIneptique** [comment]
Basically, it seems that this is called on applying unicode normalization and this leads to this

https://github.com/mittagessen/kraken/blob/10a320e03609fd97775b644686e2b379c7535f76/kraken/lib/dataset/recognition.py#L170-L179

I wonder why we should check for empty lines and most importantly raise there ?

**Comment by mittagessen** [comment]
On 22/09/29 08:42AM, Thibault Clérice wrote:
> Basically, it seems that this is called on applying unicode normalization and this leads to this
> 
> https://github.com/mittagessen/kraken/blob/10a320e03609fd97775b644686e2b379c7535f76/kraken/lib/dataset/recognition.py#L170-L179
> 
> I wonder why we should check for empty lines and most importantly raise there ?

That's fixed in master already. The difference is for the use of the
dataset. For unsupervised pretraining it doesn't matter that the line
contains a transcription or not, for training we want to filter them
out.

**Comment by PonteIneptique** [comment]
Thanks, I did not think to check on master.

Le jeu. 29 sept. 2022 à 11:37 AM, mittagessen ***@***.***> a
écrit :

**Comment by PonteIneptique** [comment]
> That's fixed in master already. The difference is for the use of the
dataset. For unsupervised pretraining it doesn't matter that the line
contains a transcription or not, for training we want to filter them
out.

I understand wanting to filter them out but should it raise an exception while rebuilding the alphabet ? (I added a small comment over your last commit).

**Comment by mittagessen** [comment]
No, it shouldn't. I screwed up the fix of the merge regression introduced by the dataset refactor in the shapely 2.0 branch.
